### PR TITLE
Poking at zombies to make them more interesting

### PR DESCRIPTION
**Pull request by ghost**
Pokes at zombies
- Adds zombies eating at corpses, this restores their health, and potentially turns the corpse into a zombie
- Adds zombie 'evolution' depending on what's happened to them (Die too much and eat too little, become something that should deal with whatever's killing you. Die too little and eat too much, there's too many corpses around so start converting them into zombies)
- 'Turned' zombie - Looks like a human, can potentially be converted back if caught in time
- 'Rotting' zombie - Somebody too far gone to bring back
- 'Necrotic' zombie - When it eats at a dead body, has a chance of converting the corpse into another zombie
- 'Crimson' zombie - Angry zombie that should tear down doors

TODO
- [x] Fix their AI so they're not so damn glutenous or pacified
- [x] Add the door smashing mechanic
- [x] Make sure everything doesn't crash

Also fixes necromancy mobs from attacking the necromancer

### CODE REVIEW

**Comment by ghost** [comment]
Okay, working on expanding this a bit more, hang on

File: code/modules/mob/living/simple_animal/hostile/necro.dm [hunk 1]
```diff
@@ -98,11 +110,205 @@
...
 
-/mob/living/simple_animal/hostile/necro/copy/ListTargets()
-	. = ..()
-	return . - creator
+	if(name == "zombie" || name == "skeleton")
+		name += " ([rand(1,1000)])"
+
+/mob/living/simple_animal/hostile/necro/zombie/Life()
+	..()
+	/*TODO
```
Comment (Skullyton) [inline code/modules/mob/living/simple_animal/hostile/necro.dm]: Now, they'll happily eat on dead corpses in the vicinity, and wander, but they just don't care about potential targets and I'm not entirely sure why

**Comment by Skullyton** [comment]
Now why that's failing to compile I'm not sure, it compiles my side

File: code/modules/mob/living/simple_animal/hostile/necro.dm [hunk 2]
```diff
@@ -98,11 +110,205 @@
...
+	First, check if the zombie can potentially evolve
+	Have the zombie move to a corpse and start chewing at it
+	If neither of these things are applicable, break down some doors
+	If none of the above, play dead
+	*/
+	if(!stat)
+		if(stance = HOSTILE_STANCE_IDLE) //Not doing anything at the time
```
Comment (JustSumBody) [inline code/modules/mob/living/simple_animal/hostile/necro.dm]: Shouldn't this be `stance == HOSTILE_STANCE_IDLE` ?

Comment (PJB3005) [inline code/modules/mob/living/simple_animal/hostile/necro.dm]: Yes

Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/necro.dm]: Oh hell, don't tell me this little thing was what was causing the AI to screwball

Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/necro.dm]: Holy crap it was, peer review saves the day again

File: code/modules/mob/living/simple_animal/hostile/necro.dm [hunk 3]
```diff
@@ -98,11 +110,219 @@
...
+	health = 150
+	can_evolve = 0
+	var/zombify_chance = 25 //Down with hardcoding
+
+/mob/living/simple_animal/hostile/necro/zombie/putrid/eat(mob/living/carbon/human/target)
+	..()
+	if(target.health < -150) //Gotta be a bit chewed on
+		visible_message("<span class='warning'>\The [target] stirs, as if it's trying to get up.</span>")
+		if(prob(zombify_chance))
+			to_chat(world, "[target] zombified by [src]")
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/necro.dm:304]: Missed one

Comment (9600bauds) [inline code/modules/mob/living/simple_animal/hostile/necro.dm:304]: HITLERS

**Comment by ghost** [comment]
So, the conflicts are because of 64x64, right?

**Comment by ghost** [comment]
Players already can barely deal with necromorphs ans zombies, A good staff of necromancy Wizard can ruin the station with just a few zombies. 
I don't like the idea of them destroying doors, what of they worked like an emag? Forced it open and break the airlock controller, or maybe like a wirejack, just forcing it open.
As much as I would love for zombies to eat corpses and make more zombies, the station would shit itself and be over-ran so fast with our playerbase. You can scream 'butcher the zombies before they regenerate' all day and only 1/20 of the players will actually bother.

**Comment by ghost** [comment]
They force them open like applying a crowbar to one you've pulsed the power of, you just have to hit said door with an ID with the same access to close it again.

That, or bolts, and hope the people on the ground didn't cause a brute to form.

The staff of necromancy makes regular zombies, which can't evolve, if it's a bother I'll make them exempt from being able to open doors as well, so they're only limited to eating

**Comment by ghost** [comment]
I fucked up, I fucked up so bad

**Pull request closed without merging**